sycl/plugins/unified_runtime/adapters/level_zero/ur_level_zero.cpp

```diff
@@ -127,6 +127,10 @@ ze_structure_type_t getZeStructureType<ze_device_memory_properties_t>() {
   return ZE_STRUCTURE_TYPE_DEVICE_MEMORY_PROPERTIES;
 }
 template <>
+ze_structure_type_t getZeStructureType<ze_device_memory_ext_properties_t>() {
+  return ZE_STRUCTURE_TYPE_DEVICE_MEMORY_EXT_PROPERTIES;

```

Comment:
this will work, although technically we should be checking if underlying driver supports this extension  ZE_DEVICE_MEMORY_PROPERTIES_EXT_NAME with zeDriverGetExtensionProperties

https://spec.oneapi.io/level-zero/latest/core/api.html?highlight=zedrivergetextensionproperties#zedrivergetextensionproperties

Reply:
what's the worst that can possibly happen? 
will zeDeviceGetMemoryProperties write null to pNext: https://github.com/intel/llvm/pull/7789/files#diff-15dd1eb076d2164bd9e87d9057f05f652a716498e8cdf5975e564c65309a0985R807
?

do I need to check it, or all parties already support this?

Reply:
@smaslov-intel since this is an extension, an L0 driver is not obligated to support it to be spec compliant. Since the plugin is only calling L0 GPU driver, and we know it has it, then no concerns, but it is just we want to be strictly correct.

Reply:
OK, I will not bother with the check then.